### DISCUSSION TITLE
Naming convention of parameters of value tuples

### DISCUSSION THREAD

**Discussion by WhiteBlackGoose** [dotnet/csharplang#5220] [Q&A] [answered]
Am I correct, that the accepted naming convention for parameters of value tuples is camelCase? If not, skip this post :P

If so, then isn't that inconsistent with
1) public property convention (PascalCase)
2) Item1, Item2 etc. default properties' names?

I used to write them all camelCase, but now I'm more inclined to write Pascal, which makes it overall more consistent with other code, but at the same time inconsistent with other value tuples. So I'm a bit lost.

Why camelCase?

**Comment by CyrusNajmabadi** [comment]
> Why camelCase?

They were designed to mimic passing around a parameter list, just wrapped into a single value. 

Note, if you want a different making style, then fell free. It's your code, do what you feel is best.

**Reply by WhiteBlackGoose** [reply]
Thanks

**Comment by jnm2** [comment, marked as answer]
This was hashed out at great length in https://github.com/dotnet/runtime/issues/27939 when the BCL added their first tuple-returning API and set the precedent to pascal case. And now, for example, StyleCop defaults to the BCL precedent.

**Reply by WhiteBlackGoose** [reply]
I'm a bit lost in that long discussion... am I right, that the overall agreement is to use PascalCase for public API, and camelCase for local use?

**Reply by jnm2** [reply]
I don't think local use was determined in that thread, but for consistency I've been using exclusively pascal case so that dotting looks the same everywhere.

**Reply by WhiteBlackGoose** [reply]
Awesome. Thanks a lot
